tiledb/sm/fragment/fragment_info.h

```diff
@@ -107,6 +107,9 @@ class FragmentInfo {
   Status get_total_cell_num(uint64_t* cell_num) const;
 
   /** Retrieves the name of the fragment with the given index. */
+  const std::string& get_fragment_name_str(uint32_t fid) const;
+
+  /** Retrieves the name of the fragment with the given index. Deprecated. */
   Status get_fragment_name(uint32_t fid, const char** name) const;
```

Comment:
We should remove this function in this PR. It's only use is for the deprecated API function, which can be simply reimplemented, albeit with a memory leak.
Having been through the pain of removing C API functions after deprecation, it is very useful that deprecated functions do not drag old implementations with them for the duration until final removal. This PR is altering this function; it would be better to remove it and alter the C API function instead.